migrator/migrations/m_189_to_m_190_vulnerability_requests_add_name/store/updated/store.go

```diff
@@ -44,15 +44,16 @@ type Store interface {
 
 // New returns a new Store instance using the provided sql instance.
 func New(db postgres.DB) Store {
-	return pgSearch.NewGenericStoreWithPermissionChecker[storeType, *storeType](
+	return pgSearch.NewGenericStore[storeType, *storeType](
```

Comment:
How about having a dedicated method so we only pass `targetResource` and it will call `GloballyScopedUpsertChecker` which can be unexported then